### PR TITLE
Constant evaluation of `newAddress`

### LINKED ISSUE #214: `newAddress()` should be const eval
## Problem

```
contract A {
    myAddress: Address = newAddress(0, 0);
    receive("set") {
        self.myAddress = sender();
    }
}
```

gives error `Cannot reduce expression to a constant Address`. In order to avoid this any `Address` type variable must be initialized in the `init()` section.

## Proposal

`newAddress()` should be const eval. This would allow more consistent code, i.e all data types (int, bool, Address) can be initiated in the declaration section of the contract.

### PR DESCRIPTION
**Pull request by Gusarich**
Closes #214 

- [x] I have updated CHANGELOG.md
- [x] I have added unit tests to demonstrate the contribution is correctly implemented
- [x] I have run all the tests locally and no test failure was reported
- [x] I did not do unrelated and/or undiscussed refactorings

also covered constants with test. for some reason there were no checks for their values.

### CODE REVIEW

File: src/types/resolveConstantValue.ts [hunk 1]
```diff
@@ -121,6 +121,23 @@ function reduceAddress(ast: ASTExpression, ctx: CompilerContext): Address {
...
                 return address;
             }
+        } else if (ast.name === "newAddress") {
+            if (ast.args.length === 2) {
+                const wc = reduceInt(ast.args[0]);
+                const addr = Buffer.from(
+                    reduceInt(ast.args[1]).toString(16),
+                    "hex",
+                );
+                if (!enabledMasterchain(ctx)) {
```
Comment (anton-trunov) [inline src/types/resolveConstantValue.ts:137]: We are not checking the workchain is either `-1` or `0` like in the code above. Why is that?

File: src/types/resolveConstantValue.ts [hunk 2]
```diff
@@ -121,6 +121,23 @@ function reduceAddress(ast: ASTExpression, ctx: CompilerContext): Address {
                 }
                 return address;
             }
+        } else if (ast.name === "newAddress") {
```
Comment (anton-trunov) [inline src/types/resolveConstantValue.ts:124]: The first and second arguments are not necessarily literals, those can be constant expressions too, so we need to recursively process them. (Even if we don't do it elsewhere)

**Review by anton-trunov — CHANGES_REQUESTED** [review]

File: src/types/resolveConstantValue.ts (hunk 2 above)
Comment (Gusarich) [reply src/types/resolveConstantValue.ts:124]: > The first and second arguments are not necessarily literals, those can be constant expressions too, so we need to recursively process them. (Even if we don't do it elsewhere)

but that actually happens on the next lines in the code. `reduceInt(ast.args[0])` and `reduceInt(ast.args[1])`.

Comment (anton-trunov) [reply src/types/resolveConstantValue.ts:124]: you are right, I misread the code, sorry

**Review by anton-trunov — APPROVED** [review]
LGTM

**Pull request merged**